File: scripts/playwright-demo.py
from playwright.sync_api import sync_playwright
import time

def run_test():
    with sync_playwright() as p:
        print("Launching browser...")
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        print("Navigating to http://localhost:5001...")
        try:
            page.goto('http://localhost:5001')
            page.wait_for_load_state('networkidle')
            
            print(f"Page title: {page.title()}")
            
            # Take screenshot of landing page
            page.screenshot(path='landing_page.png')
            print("Landing page screenshot saved.")
            
            # Find and click login button/link
            # Let's look for a link that says "Login" or has /login href
            login_link = page.locator('a[href="/login"]').first
            if login_link.is_visible():
                print("Clicking Login link...")
                login_link.click()
                page.wait_for_load_state('networkidle')
                print(f"Navigated to: {page.url}")
                page.screenshot(path='login_page.png')
                print("Login page screenshot saved.")
            else:
                print("Login link not found. Searching for buttons...")
                buttons = page.locator('button').all()
                for i, btn in enumerate(buttons):
                    print(f"Button {i}: {btn.inner_text()}")
                    if "login" in btn.inner_text().lower():
                        btn.click()
                        page.wait_for_load_state('networkidle')
                        print(f"Navigated to: {page.url}")
                        page.screenshot(path='login_page.png')
                        break
        except Exception as e:
            print(f"Error during test: {e}")
            page.screenshot(path='error_state.png')
        
        browser.close()

if __name__ == "__main__":
    run_test()

File: playwright_demo.js
import { chromium } from "playwright";
import fs from "fs";

async function runTest() {
  console.log("Launching browser...");
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  page.on("console", (msg) => console.log("BROWSER LOG:", msg.text()));
  page.on("pageerror", (err) => console.error("BROWSER ERROR:", err.message));

  console.log("Navigating to http://127.0.0.1:5001...");
  try {
    await page.goto("http://127.0.0.1:5001", { waitUntil: "load", timeout: 60000 });

    // Wait for anything to appear in #root
    console.log("Waiting for #root to have content...");
    await page.waitForFunction(
      () => {
        const root = document.getElementById("root");
        return root && root.children.length > 0;
      },
      { timeout: 30000 }
    );

    console.log(`Page title: ${await page.title()}`);

    // Wait a bit for animations/content to settle
    await page.waitForTimeout(2000);

    // Take screenshot of landing page
    await page.screenshot({ path: "landing_page.png" });
    console.log("Landing page screenshot saved.");

    // Check if we are on landing or redirected to dashboard
    const url = page.url();
    console.log(`Current URL: ${url}`);

    if (url.includes("/login") || url.endsWith(":5001/")) {
      const getStartedBtn = page
        .locator(
          'button:has-text("Get My Plan"), button:has-text("Get Started"), button:has-text("Login")'
        )
        .first();

      if (await getStartedBtn.isVisible()) {
        console.log(`Found button: ${await getStartedBtn.innerText()}`);
        await getStartedBtn.click();
        await page.waitForLoadState("load");
        console.log(`Navigated to after click: ${page.url()}`);
        await page.screenshot({ path: "after_click.png" });
      } else {
        console.log("Primary buttons not visible. Printing all buttons:");
        const buttons = await page.locator("button").all();
        for (let i = 0; i < buttons.length; i++) {
          console.log(`Button ${i}: ${await buttons[i].innerText()}`);
        }
      }
    }
  } catch (e) {
    console.error(`Error during test: ${e}`);
    const content = await page.content();
    console.log("Page content snippet:", content.substring(0, 1000));
    await page.screenshot({ path: "error_state.png" });
  } finally {
    await browser.close();
  }
}

runTest().catch(console.error);
